### DISCUSSION TITLE
Need help with running pySCENIC ctx (step 6) using ENCODE track-base database

### DISCUSSION THREAD

**Discussion by nghiemdangvo** [aertslab/SCENIC#349] [Python]
Hi, thank you for the wonderful tool to develop SCENIC. 

I am a new user and I am having some trouble trying to run the pySCENIC pipeline on CLI, while using the ENCODE track-base database instead of the motif ranking database as described in the standard workflow. This is just to refine the identified regulons that I have obtained with the previous standard workflow.

In detail, here are the details of my run (on local computer):

ChIP-seq databases obtained from the lab's website resource [https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg38/refseq_r80/tc_v1]

> encode_20190621__ChIP_seq_transcription_factor.hg38__refseq_r80-10kb_up_and_down_tss.max.genes_vs_tracks.rankings.feather

> encode_20190621__ChIP_seq_transcription_factor.hg38__refseq-r80__10kb_up_and_down_tss.max.genes_vs_tracks.rankings.feather.sha1sum.txt

Here are the running command and following parameters:

```
pyscenic ctx ./data/adj.tsv \
> ./data/encode_20190621__ChIP_seq_transcription_factor.hg38__refseq-r80__10kb_up_and_down_tss.max.genes_vs_tracks.rankings.feather \
> --annotations_fname ./data/encode_20190621__ChIP_seq_transcription_factor.hg38__refseq-r80__10kb_up_and_down_tss.max.genes_vs_tracks.rankings.feather.sha1sum.txt \
> --expression_mtx_fname ./data/S9_preprocessed_SCENIC_input.loom \
> --mode "dask_multiprocessing" \
> --output reg_chip.csv --num_workers 4 --mask_dropouts
```

After running for a few minutes, the command terminated and thrown an error as follow:

> Traceback (most recent call last):
  File "/Users/DangVo/miniconda/envs/scrnaseq/bin/pyscenic", line 8, in <module>
    sys.exit(main())
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pyscenic/cli/pyscenic.py", line 677, in main
    args.func(args)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pyscenic/cli/pyscenic.py", line 221, in prune_targets_command
    df_motifs = calc_func(
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pyscenic/prune.py", line 400, in prune2df
    return _distributed_calc(
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pyscenic/prune.py", line 247, in _distributed_calc
    motif_annotations = load_motif_annotations(
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pyscenic/utils.py", line 48, in load_motif_annotations
    df = pd.read_csv(fname, sep='\t', index_col=[1, 0], usecols=column_names)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pandas/util/_decorators.py", line 211, in wrapper
    return func(*args, **kwargs)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pandas/util/_decorators.py", line 317, in wrapper
    return func(*args, **kwargs)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 950, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 605, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 1442, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 1747, in _make_engine
    return mapping[engine](f, **self.options)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pandas/io/parsers/c_parser_wrapper.py", line 148, in __init__
    self._validate_usecols_names(usecols, self.orig_names)
  File "/Users/DangVo/miniconda/envs/scrnaseq/lib/python3.9/site-packages/pandas/io/parsers/base_parser.py", line 918, in _validate_usecols_names
    raise ValueError(
ValueError: Usecols do not match columns, columns expected but not found: ['description', '#motif_id', 'motif_similarity_qvalue', 'orthologous_identity', 'gene_name']

So my interpretation is that there is some mismatch in the formats of the expected data input, or could I have potentially not given the correct syntax for the `pyscenic ctx` command?

Computer configuration & software versions:
pySCENIC = 0.12.0 (run on conda env)
miniconda version = 4.14.0
python package version`conda list`

> # Name                    Version                   Build  Channel
adjusttext                0.7.3                    pypi_0    pypi
aiohttp                   3.8.3                    pypi_0    pypi
aiosignal                 1.2.0                    pypi_0    pypi
alabaster                 0.7.12                     py_0    conda-forge
anndata                   0.8.0              pyhd8ed1ab_1    conda-forge
aom                       3.5.0                hf0c8a7f_0    conda-forge
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
applaunchservices         0.3.0              pyhd8ed1ab_2    conda-forge
appnope                   0.1.3              pyhd8ed1ab_0    conda-forge
arboreto                  0.1.6                    pypi_0    pypi
argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0           py39h63b48b0_2    conda-forge
arpack                    3.7.0                hefb7bc6_2    conda-forge
arrow                     1.2.3              pyhd8ed1ab_0    conda-forge
astroid                   2.12.10          py39h6e9494a_0    conda-forge
async-timeout             4.0.2                    pypi_0    pypi
atomicwrites              1.4.1              pyhd8ed1ab_0    conda-forge
attrs                     22.1.0             pyh71513ae_1    conda-forge
autopep8                  1.6.0              pyhd8ed1ab_1    conda-forge
babel                     2.10.3             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
beautifulsoup4            4.11.1             pyha770c72_0    conda-forge
bedtools                  2.30.0               h0e31d98_3    bioconda
binaryornot               0.4.4                      py_1    conda-forge
biopython                 1.79             py39h63b48b0_2    conda-forge
bioservices               1.10.0             pyhd8ed1ab_0    conda-forge
black                     22.8.0           py39h6e9494a_0    conda-forge
bleach                    5.0.1              pyhd8ed1ab_0    conda-forge
blosc                     1.21.1               h97e831e_3    conda-forge
bokeh                     2.4.3                    pypi_0    pypi
boltons                   21.0.0                   pypi_0    pypi
brotli                    1.0.9                h5eb16cf_7    conda-forge
brotli-bin                1.0.9                h5eb16cf_7    conda-forge
brotlipy                  0.7.0           py39h63b48b0_1004    conda-forge
brunsli                   0.1                  h046ec9c_0    conda-forge
bzip2                     1.0.8                h0d85af4_4    conda-forge
c-ares                    1.18.1               h0d85af4_0    conda-forge
c-blosc2                  2.4.2                had5f71b_0    conda-forge
ca-certificates           2022.9.24            h033912b_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cellrank                  1.5.1              pyhdfd78af_0    bioconda
certifi                   2022.9.24          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1           py39hae9ecf2_0    conda-forge
cfitsio                   4.1.0                h2c97ad1_0    conda-forge
chardet                   5.0.0            py39h6e9494a_0    conda-forge
charls                    2.3.4                he49afe7_0    conda-forge
charset-normalizer        2.1.1              pyhd8ed1ab_0    conda-forge
click                     8.1.3            py39h6e9494a_0    conda-forge
cloudpickle               2.2.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.5              pyhd8ed1ab_0    conda-forge
colorlog                  6.7.0            py39h6e9494a_0    conda-forge
contourpy                 1.0.5            py39h92daf61_0    conda-forge
cookiecutter              2.1.1              pyh6c4a22f_0    conda-forge
cryptography              37.0.1           py39hf6deb26_0  
ctxcore                   0.2.0                    pypi_0    pypi
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cytoolz                   0.12.0           py39h701faf5_0    conda-forge
dask-core                 2022.9.2           pyhd8ed1ab_0    conda-forge
dav1d                     1.0.0                hac89ed1_1    conda-forge
dbus                      1.13.6               h811a1a6_3    conda-forge
debugpy                   1.6.3            py39hd91caee_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
diff-match-patch          20200713           pyh9f0ad1d_0    conda-forge
dill                      0.3.5.1            pyhd8ed1ab_0    conda-forge
distributed               2022.9.2                 pypi_0    pypi
docrep                    0.3.2              pyh44b312d_0    conda-forge
docutils                  0.19             py39h6e9494a_0    conda-forge
easydev                   0.12.0             pyh6c4a22f_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
expat                     2.4.9                hf0c8a7f_0    conda-forge
fa2                       0.3.5            py39h89e85a6_1    conda-forge
fftw                      3.3.10          mpi_mpich_h4a8b384_5    conda-forge
flake8                    4.0.1              pyhd8ed1ab_2    conda-forge
flit-core                 3.7.1              pyhd8ed1ab_0    conda-forge
fonttools                 4.37.4           py39ha30fb19_0    conda-forge
freetype                  2.12.1               h3f81eb7_0    conda-forge
frozendict                2.3.4                    pypi_0    pypi
frozenlist                1.3.1                    pypi_0    pypi
fsspec                    2022.8.2           pyhd8ed1ab_0    conda-forge
future                    0.18.2           py39h6e9494a_5    conda-forge
gettext                   0.19.8.1          hd1a6beb_1008    conda-forge
gevent                    21.12.0          py39hf71ec93_2    conda-forge
giflib                    5.2.1                hbcb3906_2    conda-forge
glpk                      4.65              h0f52abe_1004    conda-forge
gmp                       6.2.1                h2e338ed_0    conda-forge
greenlet                  1.1.3            py39h7a8716b_0    conda-forge
grequests                 0.6.0              pyh9f0ad1d_0    conda-forge
gseapy                    0.13.0           py39h693c9d2_0    bioconda
h5py                      3.7.0           nompi_py39hf4bc8c9_101    conda-forge
hdf5                      1.12.2          mpi_mpich_hc154f39_0    conda-forge
heapdict                  1.0.1                    pypi_0    pypi
html5lib                  1.1                pyh9f0ad1d_0    conda-forge
hypre                     2.25.0          mpi_mpich_he15861c_0    conda-forge
icu                       69.1                 he49afe7_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
igraph                    0.9.7                hacf96bb_0    conda-forge
imagecodecs               2022.9.26        py39h2af7347_0    conda-forge
imageio                   2.22.0             pyhfa7a67d_0    conda-forge
imagesize                 1.4.1              pyhd8ed1ab_0    conda-forge
importlib-metadata        4.11.4           py39h6e9494a_0    conda-forge
importlib_metadata        4.11.4               hd8ed1ab_0    conda-forge
importlib_resources       5.9.0              pyhd8ed1ab_0    conda-forge
inflection                0.5.1              pyh9f0ad1d_0    conda-forge
interlap                  0.2.7                    pypi_0    pypi
intervaltree              3.0.2                      py_0    conda-forge
ipykernel                 6.16.0             pyh736e0ef_0    conda-forge
ipython                   7.33.0           py39h6e9494a_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                8.0.2              pyhd8ed1ab_1    conda-forge
isort                     5.10.1             pyhd8ed1ab_0    conda-forge
jaraco.classes            3.2.2              pyhd8ed1ab_0    conda-forge
jedi                      0.18.1             pyhd8ed1ab_2    conda-forge
jellyfish                 0.9.0            py39h63b48b0_1    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
jinja2-time               0.2.0              pyhd8ed1ab_3    conda-forge
joblib                    1.2.0              pyhd8ed1ab_0    conda-forge
jpeg                      9e                   hac89ed1_2    conda-forge
jsonschema                4.16.0             pyhd8ed1ab_0    conda-forge
jupyter_client            7.3.5              pyhd8ed1ab_0    conda-forge
jupyter_core              4.11.1           py39h6e9494a_0    conda-forge
jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
jupyterlab_widgets        3.0.3              pyhd8ed1ab_0    conda-forge
jxrlib                    1.1                  h35c211d_2    conda-forge
keyring                   23.9.3           py39h6e9494a_0    conda-forge
kiwisolver                1.4.4            py39h7c694c3_0    conda-forge
krb5                      1.19.3               hb98e516_0    conda-forge
lazy-object-proxy         1.7.1            py39h63b48b0_1    conda-forge
lcms2                     2.12                 h577c468_0    conda-forge
leidenalg                 0.8.9            py39hfd1d529_0    conda-forge
lerc                      4.0.0                hb486fe8_0    conda-forge
libaec                    1.0.6                he49afe7_0    conda-forge
libavif                   0.10.1               he9a59b1_2    conda-forge
libblas                   3.9.0           16_osx64_openblas    conda-forge
libbrotlicommon           1.0.9                h5eb16cf_7    conda-forge
libbrotlidec              1.0.9                h5eb16cf_7    conda-forge
libbrotlienc              1.0.9                h5eb16cf_7    conda-forge
libcblas                  3.9.0           16_osx64_openblas    conda-forge
libclang                  13.0.1          default_he082bbe_0    conda-forge
libcurl                   7.85.0               h581aaea_0    conda-forge
libcxx                    14.0.6               hccf4f1f_0    conda-forge
libdeflate                1.14                 hb7f2c08_0    conda-forge
libedit                   3.1.20191231         h0678c8f_2    conda-forge
libev                     4.33                 haf1e3a3_1    conda-forge
libffi                    3.4.2                h0d85af4_5    conda-forge
libgfortran               5.0.0           10_4_0_h97931a8_25    conda-forge
libgfortran5              11.3.0              h082f757_25    conda-forge
libglib                   2.72.1               hfbcb929_0    conda-forge
libiconv                  1.16                 haf1e3a3_0    conda-forge
liblapack                 3.9.0           16_osx64_openblas    conda-forge
libllvm11                 11.1.0               h8fb7429_4    conda-forge
libllvm13                 13.0.1               h64f94b2_2    conda-forge
libnghttp2                1.47.0               h5aae05b_1    conda-forge
libopenblas               0.3.21          openmp_h429af6e_3    conda-forge
libpng                    1.6.38               ha978bb4_0    conda-forge
libpq                     14.5                 h2ca9b15_0    conda-forge
libsodium                 1.0.18               hbcb3906_1    conda-forge
libspatialindex           1.9.3                he49afe7_4    conda-forge
libsqlite                 3.39.4               ha978bb4_0    conda-forge
libssh2                   1.10.0               h47af595_3    conda-forge
libtiff                   4.4.0                hdb44e8a_4    conda-forge
libuv                     1.44.2               hac89ed1_0    conda-forge
libwebp-base              1.2.4                h775f41a_0    conda-forge
libxcb                    1.13              h0d85af4_1004    conda-forge
libxml2                   2.9.12               h7e28ab6_1    conda-forge
libxslt                   1.1.33               h1acebb3_3    conda-forge
libzlib                   1.2.12               hfd90126_4    conda-forge
libzopfli                 1.0.3                h046ec9c_0    conda-forge
llvm-openmp               14.0.4               ha654fa7_0    conda-forge
llvmlite                  0.39.1           py39had167e2_0    conda-forge
locket                    1.0.0              pyhd8ed1ab_0    conda-forge
loompy                    3.0.6                      py_0    conda-forge
lxml                      4.8.0            py39h63b48b0_2    conda-forge
lz4-c                     1.9.3                he49afe7_1    conda-forge
markupsafe                2.1.1            py39h63b48b0_1    conda-forge
matplotlib-base           3.6.0            py39hb2f573b_0    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mccabe                    0.6.1                      py_1    conda-forge
metis                     5.1.0             h2e338ed_1006    conda-forge
mistune                   2.0.4              pyhd8ed1ab_0    conda-forge
more-itertools            8.14.0             pyhd8ed1ab_0    conda-forge
mpfr                      4.1.0                h0f52abe_1    conda-forge
mpi                       1.0                       mpich    conda-forge
mpi4py                    3.1.3            py39hef65b2c_2    conda-forge
mpich                     4.0.2              hd33e60e_100    conda-forge
msgpack                   1.0.4                    pypi_0    pypi
multidict                 6.0.2                    pypi_0    pypi
multiprocessing-on-dill   3.5.0a4                  pypi_0    pypi
mumps-include             5.2.1               h694c41f_11    conda-forge
mumps-mpi                 5.2.1               h690e093_11    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
mypy_extensions           0.4.3            py39h6e9494a_5    conda-forge
mysql-common              8.0.30               hc4b2c72_1    conda-forge
mysql-libs                8.0.30               h8658499_1    conda-forge
natsort                   8.2.0              pyhd8ed1ab_0    conda-forge
nb_conda                  2.2.1            py39h6e9494a_4    conda-forge
nb_conda_kernels          2.3.1            py39h6e9494a_1    conda-forge
nbclient                  0.6.8              pyhd8ed1ab_0    conda-forge
nbconvert                 7.1.0              pyhd8ed1ab_0    conda-forge
nbconvert-core            7.1.0              pyhd8ed1ab_0    conda-forge
nbconvert-pandoc          7.1.0              pyhd8ed1ab_0    conda-forge
nbformat                  5.6.1              pyhd8ed1ab_0    conda-forge
ncls                      0.0.65           py39h7222f55_1    bioconda
ncurses                   6.3                  h96cf925_1    conda-forge
nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
networkx                  2.8.7              pyhd8ed1ab_0    conda-forge
notebook                  6.4.12             pyha770c72_0    conda-forge
nspr                      4.32                 hcd9eead_1    conda-forge
nss                       3.78                 ha8197d3_0    conda-forge
numba                     0.56.2           py39hf6b9d82_1    conda-forge
numexpr                   2.8.3                    pypi_0    pypi
numpy                     1.23.3           py39h34843a6_0    conda-forge
numpy_groupies            0.9.19             pyhd8ed1ab_0    conda-forge
numpydoc                  1.4.0              pyhd8ed1ab_1    conda-forge
openjpeg                  2.5.0                h5d0d7b0_1    conda-forge
openssl                   3.0.5                hfd90126_2    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.5.0            py39hecff1ad_0    conda-forge
pandoc                    2.19.2               h694c41f_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parmetis                  4.0.3             hb7fa8f8_1005    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
partd                     1.3.0              pyhd8ed1ab_0    conda-forge
pathspec                  0.10.1             pyhd8ed1ab_0    conda-forge
patsy                     0.5.2              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 he49afe7_0    conda-forge
petsc                     3.17.4          real_h3f2e751_101    conda-forge
petsc4py                  3.17.4          real_h89018a5_100    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.2.0            py39h4d560c1_2    conda-forge
pip                       22.2.2             pyhd8ed1ab_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
platformdirs              2.5.2              pyhd8ed1ab_1    conda-forge
pluggy                    1.0.0            py39h6e9494a_3    conda-forge
progressbar2              4.0.0              pyhd8ed1ab_0    conda-forge
prometheus_client         0.14.1             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.31             pyha770c72_0    conda-forge
psutil                    5.9.2            py39ha30fb19_0    conda-forge
pthread-stubs             0.4               hc929b4f_1001    conda-forge
ptscotch                  6.0.9                h4f3afa6_2    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pyarrow                   9.0.0                    pypi_0    pypi
pycodestyle               2.8.0              pyhd8ed1ab_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pydocstyle                6.1.1              pyhd8ed1ab_0    conda-forge
pyflakes                  2.4.0              pyhd8ed1ab_0    conda-forge
pygam                     0.8.0                      py_0    conda-forge
pygments                  2.13.0             pyhd8ed1ab_0    conda-forge
pygpcca                   1.0.3            py39h6e9494a_1    conda-forge
pylint                    2.15.3             pyhd8ed1ab_0    conda-forge
pyls-spyder               0.4.0              pyhd8ed1ab_0    conda-forge
pynndescent               0.5.7              pyh6c4a22f_0    conda-forge
pyobjc-core               8.5.1            py39h0504f80_0    conda-forge
pyobjc-framework-cocoa    8.5              py39hf3ca4cb_0    conda-forge
pyobjc-framework-coreservices 8.5              py39h701faf5_1    conda-forge
pyobjc-framework-fsevents 8.5              py39h6e9494a_1    conda-forge
pyopenssl                 22.0.0             pyhd8ed1ab_1    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pyqt                      5.12.3           py39h6e9494a_8    conda-forge
pyqt-impl                 5.12.3           py39he44290a_8    conda-forge
pyqt5-sip                 4.19.18          py39h15fb055_8    conda-forge
pyqtchart                 5.12             py39he44290a_8    conda-forge
pyqtwebengine             5.12.1           py39he44290a_8    conda-forge
pyranges                  0.0.117            pyh5e36f6f_0    bioconda
pyrle                     0.0.35           py39h7222f55_1    bioconda
pyroe                     0.6.2              pyhdfd78af_0    bioconda
pyrsistent                0.18.1           py39h63b48b0_1    conda-forge
pyscenic                  0.12.0                   pypi_0    pypi
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.9.13          hf8d34f4_0_cpython    conda-forge
python-annoy              1.17.0           py39hfd1d529_4    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.16.2             pyhd8ed1ab_0    conda-forge
python-igraph             0.9.9            py39h8c2f370_0    conda-forge
python-lsp-black          1.2.1              pyhd8ed1ab_0    conda-forge
python-lsp-jsonrpc        1.0.0              pyhd8ed1ab_0    conda-forge
python-lsp-server         1.4.1              pyhd8ed1ab_1    conda-forge
python-slugify            6.1.2              pyhd8ed1ab_0    conda-forge
python-utils              3.3.3              pyhd8ed1ab_0    conda-forge
python.app                1.4              py39h89e85a6_0    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pytoolconfig              1.2.2              pyhd8ed1ab_0    conda-forge
pytz                      2022.4             pyhd8ed1ab_0    conda-forge
pywavelets                1.3.0            py39h86b5767_1    conda-forge
pyyaml                    6.0              py39h63b48b0_4    conda-forge
pyzmq                     24.0.1           py39hed8f129_0    conda-forge
qdarkstyle                3.0.3              pyhd8ed1ab_0    conda-forge
qstylizer                 0.2.2              pyhd8ed1ab_0    conda-forge
qt                        5.12.9               h2a607e2_5    conda-forge
qtawesome                 1.1.1              pyhd8ed1ab_0    conda-forge
qtconsole                 5.3.2              pyhd8ed1ab_0    conda-forge
qtconsole-base            5.3.2              pyha770c72_0    conda-forge
qtpy                      2.2.1              pyhd8ed1ab_0    conda-forge
readline                  8.1.2                h3899abd_0    conda-forge
requests                  2.28.1             pyhd8ed1ab_1    conda-forge
requests_cache            0.4.13                     py_0    conda-forge
rope                      1.3.0              pyhd8ed1ab_0    conda-forge
rtree                     1.0.0            py39h7d0d40a_1    conda-forge
scalapack                 2.2.0                hc746714_1    conda-forge
scanpy                    1.9.1              pyhd8ed1ab_0    conda-forge
scikit-image              0.19.3           py39h0082581_1    conda-forge
scikit-learn              1.1.2            py39h8031a56_0    conda-forge
scipy                     1.9.1            py39h9488793_0    conda-forge
scotch                    6.0.9                h3da7401_2    conda-forge
scrublet                  0.2.3              pyh5e36f6f_1    bioconda
scvelo                    0.2.4              pyhdfd78af_0    bioconda
seaborn                   0.12.0               hd8ed1ab_0    conda-forge
seaborn-base              0.12.0             pyhd8ed1ab_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
session-info              1.0.0              pyhd8ed1ab_0    conda-forge
setuptools                65.4.1             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
slepc                     3.17.2          real_hb732a19_100    conda-forge
slepc4py                  3.17.2          real_h800f7ee_100    conda-forge
snappy                    1.1.9                h6e38e02_1    conda-forge
snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
sorted_nearest            0.0.37           py39h7222f55_0    bioconda
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
sphinx                    5.2.3              pyhd8ed1ab_0    conda-forge
sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
sphinxcontrib-htmlhelp    2.0.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.5              pyhd8ed1ab_2    conda-forge
spyder                    5.3.0            py39h6e9494a_0    conda-forge
spyder-kernels            2.3.3            py39h6e9494a_0    conda-forge
sqlite                    3.39.4               h9ae0607_0    conda-forge
statsmodels               0.13.2           py39h15b18c7_0    conda-forge
stdlib-list               0.7.0                      py_2    conda-forge
suds-community            1.1.2              pyhd8ed1ab_0    conda-forge
suitesparse               5.10.1               h7aff33d_1    conda-forge
superlu                   5.2.2                h1f0f902_0    conda-forge
superlu_dist              7.2.0                h4bb6bf2_0    conda-forge
tabulate                  0.8.10             pyhd8ed1ab_0    conda-forge
tbb                       2021.6.0             hb8565cd_0    conda-forge
tblib                     1.7.0                    pypi_0    pypi
terminado                 0.16.0             pyhd1c38e8_0    conda-forge
text-unidecode            1.3                        py_0    conda-forge
textdistance              4.5.0              pyhd8ed1ab_0    conda-forge
texttable                 1.6.4              pyhd8ed1ab_0    conda-forge
threadpoolctl             3.1.0              pyh8a188c0_0    conda-forge
three-merge               0.1.1              pyh9f0ad1d_0    conda-forge
tifffile                  2022.8.12          pyhd8ed1ab_0    conda-forge
tinycss2                  1.1.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h5dbffcc_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tomlkit                   0.11.5             pyha770c72_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tornado                   6.2              py39h701faf5_0    conda-forge
tqdm                      4.64.1             pyhd8ed1ab_0    conda-forge
traitlets                 5.4.0              pyhd8ed1ab_0    conda-forge
typing                    3.10.0.0           pyhd8ed1ab_0    conda-forge
typing-extensions         4.3.0                hd8ed1ab_0    conda-forge
typing_extensions         4.3.0              pyha770c72_0    conda-forge
tzdata                    2022d                h191b570_0    conda-forge
ujson                     5.5.0            py39h7a8716b_0    conda-forge
umap-learn                0.5.3            py39h6e9494a_0    conda-forge
unicodedata2              14.0.0           py39h63b48b0_1    conda-forge
unidecode                 1.3.6              pyhd8ed1ab_0    conda-forge
urllib3                   1.26.11            pyhd8ed1ab_0    conda-forge
watchdog                  2.1.9            py39h0056ad7_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
widgetsnbextension        4.0.3              pyhd8ed1ab_0    conda-forge
wrapt                     1.14.1           py39h701faf5_0    conda-forge
wurlitzer                 3.0.2              pyhd8ed1ab_2    conda-forge
xmltodict                 0.13.0             pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.9                h35c211d_0    conda-forge
xorg-libxdmcp             1.1.3                h35c211d_0    conda-forge
xz                        5.2.6                h775f41a_0    conda-forge
yaml                      0.2.5                h0d85af4_2    conda-forge
yapf                      0.32.0             pyhd8ed1ab_0    conda-forge
yarl                      1.8.1                    pypi_0    pypi
zeromq                    4.3.4                he49afe7_1    conda-forge
zfp                       1.0.0                hb9aa9fd_1    conda-forge
zict                      2.2.0                    pypi_0    pypi
zipp                      3.8.1              pyhd8ed1ab_0    conda-forge
zlib                      1.2.12               hfd90126_4    conda-forge
zlib-ng                   2.0.6                hac89ed1_0    conda-forge
zope.event                4.5.0              pyh9f0ad1d_0    conda-forge
zope.interface            5.4.0            py39h63b48b0_2    conda-forge
zstd                      1.5.2                hfa58983_4    conda-forge

Any suggestion and support is really appreciated. Thanks so much in advance.

Best,

Dang

**Comment by Polligator** [comment]
not sure if this is right, you do understand that the data you are trying to use is for SCENIC+ not SCENIC, right?
also, I'm not sure if you can use .txt file as a database.

**Reply by nghiemdangvo** [reply]
Hi! thanks for your answer, I have too also figured out that I used the wrong ChIP track database for the code. I figured out to download the proper ChIP database and it worked. It's cool! 👍
